### PR TITLE
Add Bwc test for Observability

### LINKED ISSUE #276: Add backwards compatibility tests for automation
Add backwards compatibility tests for the plugin to automate end-to-end upgrade paths for faster releases. Developer documentation for implementing bwc tests and hooking them to CI: https://github.com/opensearch-project/opensearch-plugins/blob/main/TESTING.md#backwards-compatibility-testing

### PR DESCRIPTION
**Pull request by kavithacm**
### Description

- Add bwc specific Integration Test
- Update gradle.build to run bwc tests from opensearch 1.1 to opensearch 1.2
- Included snapshot of Notebooks 1.1 and Observability 1.2

### Issues Resolved
#276 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by davidcui1225** [comment]
DCO check is failing- also I see commits from a few months ago that don't look related to the BWC testing?

File: opensearch-observability/src/test/kotlin/org/opensearch/observability/PluginRestTestCase.kt [hunk 1]
```diff
@@ -165,9 +168,13 @@ abstract class PluginRestTestCase : OpenSearchRestTestCase() {
             exception.response
         }
         if (expectedRestStatus != null) {
+//            println("response is $response")
```
Comment (joshuali925) [inline opensearch-observability/src/test/kotlin/org/opensearch/observability/PluginRestTestCase.kt:171]: could you remove these comments

**Review by joshuali925 — COMMENTED** [review]
could you try to remove the `resource/bwc/1.2.0-SNAPSHOT/opensearch-observability-1.2.0.0-SNAPSHOT.zip` file and see if it works? trying to reduce git overhead

File: opensearch-observability/src/test/kotlin/org/opensearch/observability/PluginRestTestCase.kt (hunk 1 above)
Comment (davidcui1225) [inline opensearch-observability/src/test/kotlin/org/opensearch/observability/PluginRestTestCase.kt:171]: Can we remove any commented-out lines of they are not necessary?

File: opensearch-observability/src/test/kotlin/org/opensearch/observability/PluginRestTestCase.kt [hunk 2]
```diff
@@ -63,9 +63,12 @@ abstract class PluginRestTestCase : OpenSearchRestTestCase() {
         return true
     }
 
+    open fun preserveOpenSearchIndicesAfterTest(): Boolean = false
+
     @Throws(IOException::class)
     @After
     open fun wipeAllOpenSearchIndices() {
+        if (preserveOpenSearchIndicesAfterTest()) return
```
Comment (davidcui1225) [inline opensearch-observability/src/test/kotlin/org/opensearch/observability/PluginRestTestCase.kt:71]: When does this if condition trigger?

**Comment by kavithacm** [comment]
> DCO check is failing- also I see commits from a few months ago that don't look related to the BWC testing?

Yeah those commits are not related to bwc and i'm not sure how it was included in this, but i removed those unnecessary files

File: opensearch-observability/src/test/kotlin/org/opensearch/observability/PluginRestTestCase.kt (hunk 2 above)
Comment (kavithacm) [reply opensearch-observability/src/test/kotlin/org/opensearch/observability/PluginRestTestCase.kt:71]: this condition is set true in the IT created for bwc testing, so whenever BWC IT get's triggered this condition is triggered too

File: opensearch-observability/src/test/kotlin/org/opensearch/observability/PluginRestTestCase.kt (hunk 1 above)
Comment (kavithacm) [reply opensearch-observability/src/test/kotlin/org/opensearch/observability/PluginRestTestCase.kt:171]: yeah okay. I'll remove them

Comment (kavithacm) [reply opensearch-observability/src/test/kotlin/org/opensearch/observability/PluginRestTestCase.kt:171]: yeah okay. I'll remove them

**Comment by davidcui1225** [comment]
> > DCO check is failing- also I see commits from a few months ago that don't look related to the BWC testing?
> 
> Yeah those commits are not related to bwc and i'm not sure how it was included in this, but i removed those unnecessary files

Got it - my guess is maybe some leftover local branch that contained those old commits was used at some point. Could you resolve the DCO failure?

**Comment by kavithacm** [comment]
> > > DCO check is failing- also I see commits from a few months ago that don't look related to the BWC testing?
> > 
> > 
> > Yeah those commits are not related to bwc and i'm not sure how it was included in this, but i removed those unnecessary files
> 
> Got it - my guess is maybe some leftover local branch that contained those old commits was used at some point. Could you resolve the DCO failure?

Yeah i'm trying that currently

**Pull request closed without merging**